pkg/services/alerting/notifiers/base.go

```diff
@@ -83,6 +85,11 @@ func (n *NotifierBase) ShouldNotify(ctx context.Context, context *alerting.EvalC
 		}
 	}
 
+	// Do not notify when state is OK if DisableResolvedMessage is set to true
+	if context.Rule.State == models.AlertStateOK && n.DisableResolvedMessage {
```

Comment:
There should be tests for this in `base_test.go`

Reply:
Test added, catch default for backwards compatibility.